tests/__utils__/mocks.ts

```diff
@@ -18,9 +18,9 @@ export function spyOnActions(): jest.Mocked<SearchActions> {
 
 export type RecursivePartial<T> = {
   [P in keyof T]?:
-  T[P] extends (infer U)[] ? RecursivePartial<U>[] :
+  T[P] extends ((infer U)[] | undefined) ? RecursivePartial<U>[] :
     // eslint-disable-next-line @typescript-eslint/ban-types
-    T[P] extends object ? RecursivePartial<T[P]> :
+    T[P] extends (object | undefined) ? RecursivePartial<T[P]> :
```

Comment:
what is the reason we want it to extend a union type with `undefined`? if the field `T[P]` is undefined, I imagine we wouldn't want to perform `RecursivePartial<T[P]>` again here to derive the type but rather use the type `T[P]` or `undefined` as its own type.

Reply:
the problem with the current type is that it doesn't correctly identify optional object or array types. these types don't just extend `(infer U)[]` or `object`, they also extend `undefined`, which is why I needed to extend the union. I added another check first for types that only extend `undefined` so it doesn't become `RecursivePartial<undefined>`